packages/flutter/test/rendering/mock_canvas.dart

```diff
@@ -392,6 +414,29 @@ abstract class _DrawCommandPaintPredicate extends _PaintPredicate {
   }
 }
 
+class _RectPaintPredicate extends _DrawCommandPaintPredicate {
```

Comment:
This looks a lot like the RRect one. Can you make a generic one that takes a Rect on an RRect via a generic type argument and then just have the RRect and Rect versions be subclasses with the bare minimum (probably just the constructor)?